php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java

```diff
@@ -1548,8 +1553,23 @@ private void autoCompleteConstants(final PHPCompletionResult completionResult, P
                     proposals.put(globalVariable.getName(), new PHPCompletionItem.VariableItem(globalVariable, request));
                 }
             }
-            Collection<? extends VariableName> declaredVariables = ModelUtils.filter(variableScope.getDeclaredVariables(),
-                    nameKind, request.prefix);
+
+            List<VariableName> allDeclaredVariables = new ArrayList<>(variableScope.getDeclaredVariables());
+            // for nested arrow functions
+            if (variableScope instanceof ArrowFunctionScope) {
+                Scope inScope = variableScope.getInScope();
+                while (inScope != null
```

Comment:
The `null` check is not needed, `instanceof` always returns `false` for `null`.

Reply:
You are right. I'll fix it with the next changes. Thanks!